zhaquirks/tuya/ts0601_pool_sensor.py

```diff
@@ -0,0 +1,276 @@
+    UnitOfElectricPotential,
+    UnitOfTime,
+)
+from zigpy.quirks.v2.homeassistant.number import NumberDeviceClass
+from zigpy.quirks.v2.homeassistant.sensor import SensorDeviceClass, SensorStateClass
+import zigpy.types as t
+
+from zhaquirks.tuya import TUYA_QUERY_DATA, TuyaNewManufCluster
+from zhaquirks.tuya.builder import BatterySize, TuyaQuirkBuilder
+from zhaquirks.tuya.mcu import TuyaMCUCluster
+
+CONCENTRATION_MICROGRAMS_PER_LITER: Final = "mg/L"
+
+
+class TuyaPoolManufCluster(TuyaMCUCluster):
+    """Tuya Manufacturer cluster with automatic data point refresh logic."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self._update_timer_handle = None
+        self.check_interval = 60
+        self.next_refresh_interval = 0
+        self._loop = asyncio.get_running_loop()
+        self.handle_auto_update_check_change()
+
+    def handle_auto_update_cancel(self):
+        """Auto update timer cancel."""
+        if self._update_timer_handle:
+            self._update_timer_handle.cancel()
+            self._update_timer_handle = None
+
+    def handle_auto_update_setup_next_call(self, force_new_interval=False):
+        """Auto update schedule next update."""
+        tuya_cluster = self.endpoint.device.endpoints[1].in_clusters.get(
+            TuyaMCUCluster.cluster_id, None
+        )
+        if tuya_cluster and "auto_refresh_interval" in tuya_cluster.attributes_by_name:
+            interval = tuya_cluster.get("auto_refresh_interval", 0) * 60
+            # Check for a change to auto refresh number
+            if interval != self.next_refresh_interval:
+                self.handle_auto_update_cancel()
+                self.next_refresh_interval = interval
+                force_new_interval = True
+
+        if force_new_interval and self.next_refresh_interval > 0:
+            self.debug(
+                "using refresh interval of %d minutes", self.next_refresh_interval
+            )
+            self._update_timer_handle = self._loop.call_later(
+                self.next_refresh_interval, self.handle_auto_update_timer_wrapper
+            )
+
+    def handle_auto_update_check_change(self):
+        """Auto update schedule next interval check."""
+        self.handle_auto_update_setup_next_call()
+        self._loop.call_later(self.check_interval, self.handle_auto_update_check_change)
+
+    def handle_auto_update_timer_wrapper(self):
+        """Auto update handle refresh and schedule next update."""
+        self.create_catching_task(self.handle_auto_update())
+        self.handle_auto_update_setup_next_call(force_new_interval=True)
+
+    async def handle_auto_update(self):
+        """Auto update invoke data refresh command."""
+        tuya_cluster = self.endpoint.device.endpoints[1].in_clusters[
+            TuyaMCUCluster.cluster_id
+        ]
+        self.debug("sending refresh query command")
+        await tuya_cluster.command(TUYA_QUERY_DATA)
```

Comment:
For the initial PR for this sensor, can we remove the "auto refresh" logic and the (fake) `auto_refresh_interval` entity/attribute?

After this is merged, you can add it back as-is (including the tests), but it's easier to get this merged without it, since I'll have to take a closer look when I have some time.
Users can still use the sensor quirk in the mean time by using/automating the update button entity.